src/sdg/producten/tests/test_update_catalogs.py

```diff
@@ -1,27 +1,327 @@
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+        ref_product = ReferentieProductFactory.create(
+            catalogus=ref_catalog,
+        )
+        ref_product_versie = ProductVersieFactory.create(
+            product=ref_product,
+            publicatie_datum=date.today() - timedelta(days=1),
+            versie=1,
+        )
+        LocalizedProductFactory.create(
+            product_versie=ref_product_versie,
+            taal="nl",
+            product_titel_decentraal="Original title",
+            specifieke_tekst="Original text",
+        )
+
+        LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=True,
+            organisatie__owms_end_date=None,
+        )
+
+        call_command("update_catalogs")
+
+        generated_product = producten_table.get(~Q(pk=ref_product.pk))
+        generated_product_versie = producten_versies_table.filter(
+            product=generated_product
+        )
+        generated_localized_product = localized_producten_table.filter(
+            taal="nl", product_versie=generated_product_versie.first()
+        )
+
+        self.assertEqual(
+            generated_localized_product.first().product_titel_decentraal,
+            "Original title",
+        )
+        self.assertEqual(
+            generated_localized_product.first().specifieke_tekst, "Original text"
+        )
+
+        # new reference product version and localized product
+        ref_product_versie_2 = ProductVersieFactory.create(
+            product=ref_product,
+            publicatie_datum=date.today(),
+            versie=2,
+        )
+        ref_updated_localized_product_nl = LocalizedProductFactory.create(
+            product_versie=ref_product_versie_2,
+            taal="nl",
+            product_titel_decentraal="Updated reference title",
+            specifieke_tekst="Updated reference text",
+        )
+
+        # new generated product version and localized product
+        generated_product_versie.update(
+            gewijzigd_op=date.today() + timedelta(days=1),
+            publicatie_datum=date.today() + timedelta(days=1),
+        )
+        generated_localized_product.update(
+            product_titel_decentraal="Updated generated title",
+            specifieke_tekst="Updated generated text",
+        )
+
+        call_command("update_catalogs")
+        generated_updated_localized_product = (
+            localized_producten_table.filter(
+                taal="nl", product_versie=generated_product_versie.first()
+            )
+            .exclude(pk=ref_updated_localized_product_nl.pk)
+            .first()
+        )
+
+        self.assertEqual(
+            generated_updated_localized_product.product_titel_decentraal,
+            "Updated generated title",
+        )
+        self.assertEqual(
+            generated_updated_localized_product.specifieke_tekst,
+            "Updated generated text",
+        )
```

Comment:
But, the text should NOT be updated, so this proofs its not working... but tip: it's your test thats wrong :)